panelapp/panelapp_utils.py

```diff
@@ -0,0 +1,146 @@
+        ) for gene_id in genes_by_id.keys()
+    }
+    created_locuslistgenes = SeqrLocusListGene.objects.bulk_create(locuslistgenes.values(), batch_size=10000)
+    created_locuslistgenes_by_id = {lg.gene_id: lg for lg in created_locuslistgenes}
+
+    palocuslistgenes = {
+        gene_id: _create_pa_locus_list_gene(created_locuslistgenes_by_id[gene_id], panel_genes_by_id[gene_id])
+        for gene_id in genes_by_id.keys()
+    }
+    PaLocusListGene.objects.bulk_create(palocuslistgenes.values(), batch_size=10000)
+
+
+def _create_pa_locus_list_gene(seqr_locus_list_gene, panel_gene_json):
+    del panel_gene_json['panel']  # No need to keep panel info
+    del panel_gene_json['gene_data']  # No need to keep gene info as Seqr already maintain info
+    result = PaLocusListGene(
+        seqr_locus_list_gene=seqr_locus_list_gene,
+        confidence_level=panel_gene_json.get('confidence_level'),
+        biotype=panel_gene_json.get('biotype') or None,
+        penetrance=panel_gene_json.get('penetrance') or None,
+        mode_of_pathogenicity=panel_gene_json.get('mode_of_pathogenicity') or None,
+        mode_of_inheritance=panel_gene_json.get('mode_of_inheritance') or None,
+        raw_data=panel_gene_json,
+    )
+
+    return result
+
+
+def _get_all_panels(panels_url, all_results):
+    resp = requests.get(panels_url)
+    resp_json = resp.json()
+    curr_page_results = [r for r in resp_json.get('results', []) if r.get('stats', {}).get('number_of_genes', 0) > 0]
+    all_results += curr_page_results
+
+    next_page = resp_json.get('next', None)
+    if next_page is None:
+        return all_results
+    else:
+        return _get_all_panels(next_page, all_results)
+
+
+def _get_all_genes_for_panel(panel_genes_url, all_results):
+    resp = requests.get(panel_genes_url)
+    resp_json = resp.json()
+    all_results += resp_json.get('results', [])
+
+    next_page = resp_json.get('next', None)
+    if next_page is None:
+        return all_results
+    else:
+        return _get_all_genes_for_panel(next_page, all_results)
+
+
+def _create_or_update_locus_list_from_panel(user, panel_json):
+    panel_app_id = panel_json.get('id')
+    existing = _safe_get_locus_list(panel_app_id)
+
+    if existing:
+        update_model_from_json(existing.seqr_locus_list, {
+            'name': panel_json['name']
+        }, user)
+        update_model_from_json(existing, {
+            'disease_group': panel_json.get('disease_group') or None,
+            'disease_sub_group': panel_json.get('disease_sub_group') or None,
+            'status': panel_json.get('status') or None,
+            'version': panel_json.get('version') or None,
+            'version_created': panel_json.get('version_created') or None,
+            'raw_data': panel_json,
+        }, user)
+
+        return existing
+    else:
+        seqr_locus_list = create_model_from_json(SeqrLocusList, {
+            'name': panel_json['name'],
+            'description': None,
```

Comment:
It probably would be clearer for users if the description indicated that this was imported from panel app. Either that or wherever you show the description in the UI, show some additional panel app data to make it clearer

Reply:
Yes agree, it'll be more consistent with requests like [Require descriptions for new gene lists](https://github.com/broadinstitute/seqr/issues/1983).  There are a few fields from `PaLocusList` I could use to populate the description but we have yet to agree on how PanelApp gene lists should be displayed.  See answer below regarding UI.